lib/systeminfo/dns.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package systeminfo
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/gravitational/gravity/lib/storage"
+	"github.com/gravitational/trace"
+)
+
+func ResolvFromFile(filename string) (*storage.ResolvConf, error) {
+	path, err := filepath.EvalSymlinks(filename)
+	if err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	defer f.Close()
+	cfg, err := ResolvFromReader(f)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return cfg, nil
+}
+
+func ResolvFromReader(rdr io.Reader) (*storage.ResolvConf, error) {
+	// initialize with some defaults
+	resolv := &storage.ResolvConf{
+		Ndots:    1,
+		Timeout:  5,
+		Attempts: 2,
+	}
+
+	scanner := bufio.NewScanner(rdr)
+	for scanner.Scan() {
+		if err := scanner.Err(); err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		line := scanner.Text()
+
+		// strip comments
+		line = strings.Split(line, ";")[0]
+		line = strings.Split(line, "#")[0]
+
+		f := strings.Fields(line)
+		if len(f) < 1 {
+			continue
+		}
+
+		switch f[0] {
+		case "nameserver":
+			if len(f) > 1 && len(resolv.Servers) < 3 { // limit number of servers
```

Comment:
Does it make sense to enforce the same restriction as the [resolv.conf](http://man7.org/linux/man-pages/man5/resolv.conf.5.html) itself or can it use more than 3?

Reply:
It's a good point... this is the behaviour we had in planet, and I simple maintained it.

I'll update it though, as I think the argument can be made that this should parse and represent the file as it exists on the system, and not the limitations of that system.